### Wednesday 25th -- Classes from 9:30am to 17:30pm

## Module 3 -- GWAS

This module focuses on main analyses for GWAS.

### Lecture 3

gene chips, HapMap, 1000 genomes project

#### Practical aspects of GWAS

QC-HWE, call rates, MAF

Genotype imputation, imputation quality

Multiple testing, FDR, q-value

Discovery, replication studies

Report of results and GSEA

Prediction

### Lab 3

#### Linux

File formats: .gz, .tar.gz, .bz2

Data formats: .bim/.bed/.fam, .bgen, .vcf, .sumstats

tar, gzip -d -c, zcat

awk/sed

head/tail/find

cut/paste

sort

join

Bash programming, looping and seq, branching, function, Mixing programming, e.g., <<<

Multithread and parallel computing: parallel, sge

sftp/ssh, wget

Specific software

qctool, bcftools, vcftools, PLINK2, IMPUTE/MACH, SNPTEST, QUICKTEST, METAL

#### R and Bioconductor packages

Rscript

GenAssoc, snpMatrix, snpStats, GenABEL, GWAStools, QCtools, multtest

### Exercises

**TASKS**

**Suggested reading**

* Papers/books

Manolio (2008), WTCCC (2007), Stram (2014).

* Websites

[Genome-wide association study](https://en.wikipedia.org/wiki/Genome-wide_association_study) is a wikipedia entry.
